### ISSUE TITLE
Incorrect statement about pseudo-elements needing to come last in a selector

### ISSUE THREAD

**Issue by m-akinc** [mdn/content#32971]
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/CSS/Pseudo-elements

### What specific section or headline is this issue about?

Syntax

### What information was incorrect, unhelpful, or incomplete?

> The pseudo-element must appear after all the other components in the [complex](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_selectors/Selector_structure#complex_selector) or [compound](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_selectors/Selector_structure#compound_selector) selector in which it appears. For example, you can select a paragraph's first line using `p::first-line` but not the first-line's children or a hovered first line. So both `p::first-line > *` and `p::first-line:hover` are invalid.

This is not true for all pseudo-elements. `::part()` supports appending pseudo-classes like `:hover`. This is even shown in the [`::part` documentation examples](https://developer.mozilla.org/en-US/docs/Web/CSS/::part#css). I'm not sure if there are other pseudo-elements besides `::part()` that support this.

### What did you expect to see?

At least call out any exceptions (like `::part()`).

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/css/pseudo-elements`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/CSS/Pseudo-elements
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/css/pseudo-elements/index.md
* Last commit: https://github.com/mdn/content/commit/ceb7507408b15730166596ea3b2de282c4edae1f
* Document last modified: 2023-07-26T12:42:34.000Z

</details>

**Comment by Josh-Cena** [comment]
See [spec](https://drafts.csswg.org/selectors/#pseudo-element-states):

> Certain [pseudo-elements](https://drafts.csswg.org/selectors/#pseudo-element) may be immediately followed by any combination of certain [pseudo-classes](https://drafts.csswg.org/selectors/#pseudo-class), in which case the pseudo-element is represented only when it is in the corresponding state. This specification allows any pseudo-element to be followed by any combination of the [logical combination pseudo-classes](https://drafts.csswg.org/selectors/#logical-combination-pseudo-classes) and the [user action pseudo-classes](https://drafts.csswg.org/selectors/#user-action-pseudo-class). Other specifications may allow additional pseudo-classes to be attached to particular pseudo-elements. Combinations that are not explicitly allowed are [invalid selectors](https://drafts.csswg.org/selectors/#invalid-selector).
>
> NOTE: The [logical combination pseudo-classes](https://drafts.csswg.org/selectors/#logical-combination-pseudo-classes) pass any restrictions on validity of selectors at their position to their arguments.
>
> For example, since the [:hover](https://drafts.csswg.org/selectors/#hover-pseudo) pseudo-class specifies that it can apply to any pseudo-element, ::first-line:hover will match when the first line is hovered. However, since neither [:focus](https://drafts.csswg.org/selectors/#focus-pseudo) nor [::first-line](https://drafts.csswg.org/css-pseudo-4/#selectordef-first-line) define that :focus can apply to ::first-line, the selector ::first-line:focus will never match anything.
Notice that ::first-line:hover is very different from :hover::first-line, which matches the first line of any originating element that is hovered! For example, :hover::first-line also matches the first line of a paragraph when the second line of the paragraph is hovered, whereas ::first-line:hover only matches if the first line itself is hovered.

**Comment by m-akinc** [comment]
> So it seems that `p::first-line:hover` is already valid.

Yes, it turns out the MDN doc directly contradicts the spec.

**Comment by estelle** [comment]
> So it seems that `p::first-line:hover` is already valid.

tested. invalid. The entire block fails in FF, Safari, Edge (and other chromium browsers).

See https://codepen.io/estelle/pen/WNBXQNg

**Comment by Josh-Cena** [comment]
It's possible that none of the browsers implement this yet. The last issue I can find is https://github.com/w3c/csswg-drafts/issues/7085.

**Comment by m-akinc** [comment]
> tested. invalid. The entire block fails in FF, Safari, Edge (and other chromium browsers).

So at the moment, the documentation is correct that `p::first-line:hover` is invalid, but it is still incorrect that "The pseudo-element must appear after all the other components in the complex or compound selector in which it appears." `p::part(foo):hover` is valid (checked FF and Chrome). So at least `::part()` supports appending pseudo-classes.

**Comment by estelle** [comment]
> So at the moment, the documentation is correct that `p::first-line:hover` is invalid, but it is still incorrect that "The pseudo-element must appear after all the other components in the complex or compound selector in which it appears." `p::part(foo):hover` is valid (checked FF and Chrome). So at least `::part()` supports appending pseudo-classes.

Yes. Unfortunately, this isn't the simple edit that I thought it would be. We can't just remove "not supported", we have to figure out which elements support pseudo and which don't. It would also be helpful to have a timeline of intended support. Like, is it part of baseline?

**Comment by OnkarRuikar** [comment]
We can wait till browsers implement the pseudo-classing pseudo-elements feature https://issues.chromium.org/issues/40824273 and https://bugzilla.mozilla.org/show_bug.cgi?id=1122965 .

**Comment by estelle** [comment]
There is some support exception for state pseudo-classes AFTER the pseudo-element.

the example in
[`::part(namedPart):hover`](https://developer.mozilla.org/en-US/docs/Web/CSS/::part) does this (and it works).

I also created a codepen:
https://codepen.io/estelle/pen/YzoQWbN

In that codepen, I included an example with two selectors -- each with a pseudo-element  followed by :hover -- in a selector list. The `p::after:hover` invalidates the selector list. 

The pseudo-elements page should include a note about this, but i don't know what other exceptions are out there.

**Comment by estelle** [comment]
Two years without much change.

Currently `::part(x):hover` is supported but `p::first-letter:hover` is invalid

Re-tested with https://codepen.io/estelle/pen/MYbGbXP. It will have a red, crossed out h4 if `p::first-letter:hover` is supported. It is currently invalid and invalidates the entire selector block.

The following have support (this list may  not be exhaustive)

- ::cue, 
- ::details-content
- ::file-selector-button,
- ::picker(select),
- ::scroll-button(*),
- ::scroll-marker,
- ::scroll-marker-group
